src/Approximations/overapproximate.jl

```diff
@@ -333,35 +333,73 @@ function overapproximate(lm::LinearMap{N, <:AbstractZonotope},
 end
 
 """
-    overapproximate(X::LazySet{N}, dir::AbstractDirections) where {N}
+    overapproximate(X::LazySet{N}, dir::AbstractDirections; [prune]::Bool=true) where {N}
 
-Overapproximating a set with template directions.
+Overapproximate a (possibly unbounded) set with template directions.
 
 ### Input
 
-- `X`   -- set
-- `dir` -- (concrete) direction representation
+- `X`     -- set
+- `dir`   -- (concrete) direction representation
+- `prune` -- (optional, default: `true`) flag for removing redundant
+             constraints in the end
 
 ### Output
 
 A polyhedron overapproximating the set `X` with the directions from `dir`.
-If the directions are known to be bounded, the result is an `HPolytope`,
-otherwise the result is an `HPolyhedron`.
+The overapproximation is computed using support functions. If the obtained set is
+bounded, the result is an `HPolytope`. Otherwise the result is an `HPolyhedron`.
 """
-function overapproximate(X::LazySet{N}, dir::AbstractDirections{N, VN}) where {N, VN}
-    halfspaces = Vector{LinearConstraint{N, VN}}()
-    sizehint!(halfspaces, length(dir))
-    T = isbounding(dir) ? HPolytope : HPolyhedron
-    H = T(halfspaces)
+function overapproximate(X::LazySet{N}, dir::AbstractDirections{N, VN}; prune::Bool=true) where {N, VN}
+    n = dim(X)
+    H = Vector{LinearConstraint{N, VN}}()
+    sizehint!(H, length(dir))
+
     for d in dir
-        addconstraint!(H, LinearConstraint(d, ρ(d, X)))
+        sf = ρ(d, X)
+        if !isinf(sf)
+            push!(H, LinearConstraint(d, sf))
+        end
+    end
+    if prune
+        remove_redundant_constraints!(H) || throw(ArgumentError("unable to remove redundant constraints"))
+    end
+
+    # if the input is bounded and the directions are bounding => output is bounded
+    # otherwise, check boundedness of the output
+    if (isbounded(X) && isbounding(dir)) || _isbounded_stiemke(HPolyhedron(P))
+        return dim(X) == 2 ? HPolygon(H) : HPolytope(H, check_boundedness=false)
+    else
+        return HPolyhedron(P)
     end
-    return H
 end
 
 # alias with HPolytope type as second argument
-function overapproximate(X::LazySet{N}, ::Type{<:HPolytope}, dirs::AbstractDirections) where {N}
-    return overapproximate(X, dirs)
+function overapproximate(X::LazySet{N}, ::Type{<:HPolytope}, dirs::AbstractDirections{N}; prune::Bool=true) where {N}
+    P = overapproximate(X, dirs, prune=prune)
+    isa(P, HPolytope) || throw(ArgumentError("can't overapproximate with an `HPolytope` " *
+                                "because the set is unbounded; try using an `HPolyhedron`"))
```

Comment:
This does not work anymore because you can also receive an `HPolytope` now. I guess this change is also responsible for the latest error.

Reply:
I am not sure about changing that check because the method has `HPolytope` as return type listed.